content/docs/state-and-lifecycle.md

```diff
@@ -409,41 +409,41 @@ Then you can update them independently with separate `setState()` calls:
+Scalanie jest płytkie (ang. *shallow*), tzn. `this.setState({comments})` nie zmieni `this.state.posts`, lecz całkowicie nadpisze wartość `this.state.comments`.
 
-## The Data Flows Down {#the-data-flows-down}
+## Dane płyną z góry na dół {#the-data-flows-down}
 
-Neither parent nor child components can know if a certain component is stateful or stateless, and they shouldn't care whether it is defined as a function or a class.
+Ani komponenty-rodzice, ani ich dzieci nie wiedzą, czy jakiś komponent ma stan, czy też nie. Nie powinny się również przejmować tym, czy jest on funkcyjny, czy klasowy.
 
-This is why state is often called local or encapsulated. It is not accessible to any component other than the one that owns and sets it.
+Właśnie z tego powodu stan jest nazywany lokalnym lub enkapsulowanym. Nie mają do niego dostępu żadne komponenty poza tym, który go posiada i modyfikuje.
 
-A component may choose to pass its state down as props to its child components:
+Komponent może zdecydować się na przekazanie swojego stanu w dół struktury poprzez atrybuty jego komponentów potomnych:
 
 ```js
-<h2>It is {this.state.date.toLocaleTimeString()}.</h2>
+<h2>Aktualny czas: {this.state.date.toLocaleTimeString()}.</h2>
 ```
 
-This also works for user-defined components:
+To samo tyczy się komponentów własnych:
 
 ```js
 <FormattedDate date={this.state.date} />
 ```
 
-The `FormattedDate` component would receive the `date` in its props and wouldn't know whether it came from the `Clock`'s state, from the `Clock`'s props, or was typed by hand:
+Komponent `FormattedDate` otrzyma `date` jako atrybut i nie będzie w stanie rozróżnić, czy pochodzi on ze stanu lub jednego z atrybutów komponentu `Clock`, czy też został przekazany bezpośrednio przez wartość:
 
 ```js
 function FormattedDate(props) {
-  return <h2>It is {props.date.toLocaleTimeString()}.</h2>;
+  return <h2>Aktualny czas: {props.date.toLocaleTimeString()}.</h2>;
 }
 ```
 
-[**Try it on CodePen**](http://codepen.io/gaearon/pen/zKRqNB?editors=0010)
+[**Przetestuj kod na CodePen**](http://codepen.io/gaearon/pen/zKRqNB?editors=0010)
 
-This is commonly called a "top-down" or "unidirectional" data flow. Any state is always owned by some specific component, and any data or UI derived from that state can only affect components "below" them in the tree.
+Taki przepływ danych nazywany jest powszechnie jednokierunkowym (ang. *unidirectional*) lub "z góry na dół" (ang. *top-down*). Stan jest zawsze własnością konkretnego komponentu i wszelkie dane lub części UI powstałe w oparciu o niego mogą wpłynąć jedynie na komponenty znajdujące się "poniżej" w drzewie.
 
-If you imagine a component tree as a waterfall of props, each component's state is like an additional water source that joins it at an arbitrary point but also flows down.
+Wyobraź sobie, że drzewo komponentów to wodospad atrybutów. Stan każdego z komponentów to dodatkowe źródło wody, które go zasila, jednocześnie spadając w dół wraz z resztą wody.
```

Comment:
```suggestion
Wyobraź sobie, że drzewo komponentów to wodospad atrybutów, stan każdego z komponentów to dodatkowe źródło wody, które łączy się z nim w dowolnym punkcie, jednocześnie spadając w dół wraz z resztą wody.
```

Reply:
To ja bym w takim razie dodał jeszcze "a" przed "stan".

Reply:
W takim razie dałbym jeszcze "a" przed "stan".